yt/utilities/tests/test_interpolators.py

```diff
@@ -143,3 +146,33 @@ def test_get_vertex_centered_data():
         ) in str(w[-1].message)
     assert_array_equal(vec_list[("gas", "density")], vec_str)
     assert_array_equal(vec_list[("gas", "density")], vec_tuple)
+
+
+def test_replace_nonperiodic_with_extrap():
+    for nbuf in range(3):
+        # Initialize domain information
+        nx = np.array([8, 16, 32])
+        dx = 1.0
+        lo = np.zeros((3), dtype="float64") + dx / 2
+        hi = lo + (nx - 1) * dx
+
+        # Fill field with analytical linear function: f(x,y,z) = x + y + z
+        # excluding the buffer region
+        x, y, z = np.mgrid[
+            lo[0] : hi[0] : nx[0] * 1j,
+            lo[1] : hi[1] : nx[1] * 1j,
+            lo[2] : hi[2] : nx[2] * 1j,
+        ]
+        func = x + y + z
+        field = np.zeros(nx, dtype="float64")
+        field[nbuf : nx[0] - nbuf, nbuf : nx[1] - nbuf, nbuf : nx[2] - nbuf] = func[
+            nbuf : nx[0] - nbuf, nbuf : nx[1] - nbuf, nbuf : nx[2] - nbuf
+        ]
+
+        # Use fix_nonperiodic function to extrapolate data to buffer region
```

Comment:
seems like you missed a renamed method in this comment

Reply:
good catch!